src/robots/mecanum.cc

```diff
@@ -15,7 +15,28 @@ namespace BoBRobotics {
 namespace Robots {
 Mecanum::Mecanum(const char *path, bool alternativeWiring)
   : m_Serial(path), m_AlternativeWiring(alternativeWiring)
-{}
+{
+    auto tty = m_Serial.getAttributes();
```

Comment:
most of this stuff is pretty boilerplatey - it seems a bit of a shame to expose all this posix horror t all users of the class.

Reply:
Yeah, I know :-(. The problem is that the two users of the class (``Robots::Mecanum`` and ``GPSReader``) set a bunch of different options besides and I wasn't sure which were the ones that I could safely get rid of without breaking things.

Reply:
I suspect all that's different between the two devices (and 99.9% of others) is the baud rate, the number of data and stop bits and the parity. Looking at the code (which I suspect I copy-and-pasted), as I memset the whole structure to zero, the "non-canonical" flag setting has no effect. Sounds like the VMIN and VTIME business (http://unixwiz.net/techtips/termios-vmin-vtime.html) might be worth exposing as well though

Reply:
Actually, remembering more and googling:
"Canonical Mode: In canonical mode, read on the serial port will not return until a new line, EOF or EOL character is received by the Linux Kernel. ... Non-Canonical Mode: In non-canonical mode, read on the serial port will not wait until the new line. Number of bytes read depends on the MIN and TIME settings."

So I suspect the mecanum should be non-canonical and the GPS canonical so...maybe...just exposing all the settings is best